diff -rupN a/kernel/conftest.sh b/kernel/conftest.sh
--- a/kernel/conftest.sh	2019-10-12 19:20:18.000000000 +0200
+++ b/kernel/conftest.sh	2019-10-29 16:20:42.504362304 +0100
@@ -70,7 +70,7 @@ translate_and_compile_header_files() {
         CODE="#include <$file>"
 
         echo "$CODE" > conftest_headers$$.c
-        $CC $CFLAGS -c conftest_headers$$.c > conftest_headers$$.log 2>&1
+        LC_ALL=C $CC $CFLAGS -c conftest_headers$$.c > conftest_headers$$.log 2>&1
 
         if [ -f conftest_headers$$.o ]; then
             rm -f conftest_headers$$.o
